### PR TITLE
Revert "Ignore src folder in published packages"

### PR DESCRIPTION
**Pull request by NathanJolly**
### Background

https://github.com/Shopify/pos-next-react-native/issues/26233

tl;dr: Typescript is not resolving prop types correctly after this merged.

@accet-shop and @rtymchyk, @heltisace suggested that this may mess with Stocky tests in some way. If we can find a way to fix the typescript stuff without reverting this, that would be a much better solution. However, we do need to get the typescript working by the release of 1.3.0, which is scheduled for Aug 9.

### Solution

We spent a lot of time trying to fix this without reverting these changes since they are consistent with other packages in this repository.

### 🎩

Build `retail-ui-extensions-react` and verify that the typescript declaration files can resolve props of components correctly.

### CODE REVIEW

**Comment by NathanJolly** [comment]
So when someone installs `retail-ui-extensions-react` in their project, it'll look like this:
image
